HeatStroke/StateMachine/StateMachine.cpp

```diff
@@ -165,7 +165,8 @@ void HeatStroke::StateMachine::Update(const float p_fDelta, const bool m_bUpdate
 	if (m_bUpdateStack)
 	{
 		// Update entire stack from bottom to top
-		StateStack::iterator it = m_mStateStack.begin(), end = m_mStateStack.end();
+		StateStack mLocalCopy(m_mStateStack);
+		StateStack::iterator it = mLocalCopy.begin(), end = mLocalCopy.end();
```

Comment:
This doesn't seem like it should work. You're making and updating a local copy which then just immediately goes out of scope. Can you help me with more detail as to why this needs to be like this and why it works?